sklearn/utils/_array_api.py

```diff
@@ -187,10 +195,49 @@ def _union1d(a, b, xp):
 
 
 def isdtype(dtype, kind, *, xp):
-    """Returns a boolean indicating whether a provided dtype is of type "kind".
+    """Return a boolean indicating whether a provided dtype is of type "kind".
 
     Included in the v2022.12 of the Array API spec.
     https://data-apis.org/array-api/latest/API_specification/generated/array_api.isdtype.html
+
+    Parameters
+    ----------
+    dtype : dtype
+        The input dtype (e.g. ``np.float32``, ``torch.int16``, ``xp.complex64``).
+
+    kind : str or dtype or tuple[str | dtype]
+        Data type kind.
+
+        - If ``kind`` is a dtype, then a boolean is returned if ``dtype`` is equal to
+          ``dtype``.
```

Comment:
```suggestion
        - If ``kind`` is a dtype, then ``dtype`` is compared to ``kind``.
```
I've tried to make a suggestion for an improvement that explains that if you pass a dtype object for `kind`, then `dtype` is compared exactly to it. And the result is whether `kind` and `dtype` match. However I can't find a way to do it in one sentence :-/ Maybe someone else has a brain wave.

Right now we need to edit the current sentence because it says you receive a boolean if `dtype` and `dtype` are equal. Which is probably not what we are trying to say :D